src/System.Private.CoreLib/shared/System/ComponentModel/DefaultValueAttribute.cs

```diff
@@ -64,13 +65,21 @@ bool TryConvertFromInvariantString(Type typeToConvert, string stringValue, out o
                     if (s_convertFromInvariantString == null)
                     {
                         Type typeDescriptorType = Type.GetType("System.ComponentModel.TypeDescriptor, System.ComponentModel.TypeConverter", throwOnError: false);
-                        Volatile.Write(ref s_convertFromInvariantString, typeDescriptorType == null ? new object() : Delegate.CreateDelegate(typeof(Func<Type, string, object>), typeDescriptorType, "ConvertFromInvariantString", ignoreCase: false));
+                        MethodInfo mi = typeDescriptorType?.GetMethod("ConvertFromInvariantString", BindingFlags.NonPublic | BindingFlags.Static);
+                        Volatile.Write(ref s_convertFromInvariantString, mi == null ? new object() : mi.CreateDelegate(typeof(Func<Type, string, object>)));
                     }
 
                     if (!(s_convertFromInvariantString is Func<Type, string, object> convertFromInvariantString))
                         return false;
 
-                    conversionResult = convertFromInvariantString(typeToConvert, stringValue);
+                    try
+                    {
+                        conversionResult = convertFromInvariantString(typeToConvert, stringValue);
+                    }
+                    catch
```

Comment:
Would it make sense to only catch specific exceptions here?

Reply:
This is called from a top level catchall already.

The advantage of catching everything at this level is that we will try to run the fallback that might be able to fix things in more cases. The top level catchall makes things pretty undiagnosable either way. Making things work in more cases feels like an improvement, unless there's concerns about the fallback implementation.

Reply:
ok